src/partialBind.test-d.ts

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable unicorn/consistent-function-scoping */
+import { partialBind } from "./partialBind";
+
+describe("simple case (all required, no rest params)", () => {
+  const fn = (x: number, y: number, z: number | string): string =>
+    `${x}, ${y}, and ${z}`;
+
+  test("should correctly type 0 partial args", () => {
+    expectTypeOf(partialBind(fn, [])).toEqualTypeOf<
+      (x: number, y: number, z: number | string) => string
+    >();
+  });
+
+  test("should correctly type 1 partial arg", () => {
+    expectTypeOf(partialBind(fn, [1])).toEqualTypeOf<
+      (y: number, z: number | string) => string
+    >();
+  });
+
+  test("should correctly type all partial args", () => {
+    expectTypeOf(partialBind(fn, [1, 2, 3])).toEqualTypeOf<() => string>();
+  });
+
+  test("should allow refined types", () => {
+    expectTypeOf(partialBind(fn, [1, 2, "a"])).toEqualTypeOf<() => string>();
+  });
```

Comment:
It looks like these tests are the same case, but they have different names. I think I'm missing something...